mobile/yggmail/yggmail.go

```diff
@@ -234,13 +288,26 @@ func (ym *Yggmail) Stop() {
 		ym.imapServer = nil
 	}
 
-	ym.CloseDatabase()
+	if ym.transport != nil {
+		ym.transport.Stop()
+		ym.transport = nil
+	}
+	ym.closeDatabase()
+
+	ym.setState(Stopped)
 }
 
-func (ym *Yggmail) sendError(errorId int, format string, a ...interface{}) {
+func (ym *Yggmail) handleError(errorId int, format string, a ...interface{}) {
```

Comment:
As a suggestion for errors wrapping:

You can change this to 
```Go
func (ym *Yggmail) handleError(errorId int, err error) {
```
And call this as:
```Go
ym.handleError(ErrCode, xerrors.Errorf("Error description with some useful info: %w", err))
```
And as a result, you will get a detailed error object with an optional stack trace.

To print a stacktrace with all information inside your error you can use `"%v+"` placeholder, like this:
```Go
ym.Logger.LogError(errorId, fmt.Sprintf("[  Yggmail  ] %+v", err))
```

Reply:
sounds good to me. This is probably a change that should be done in the overall project, for consistency and better debugging capabilities.

Reply:
Let's discuss that change on the upstream repository.